### PR TITLE
pb-3312: Added steps to delete the resourceexport CR, when restore fails with timeout value.

### PR DESCRIPTION
**Pull request by siva-portworx**
**What this PR does / why we need it**:
```
pb-3312: Added steps to delete the resourceexport CR, when restore fails
    with timeout value.
```

**Does this PR change a user-facing CRD or CLI?**:
no

**Is a release note needed?**:
no

**Does this change need to be cherry-picked to a release branch?**:
no
Testing:
Tested restore after removing the resource.json file and checked that RE, Job and job pods are getting deleted.

### CODE REVIEW

File: pkg/applicationmanager/controllers/applicationrestore.go [hunk 1]
```diff
@@ -1695,6 +1695,22 @@ func (a *ApplicationRestoreController) cleanupRestore(restore *storkapi.Applicat
 			return fmt.Errorf("cancel restore: %s", err)
 		}
 	}
+	// Directly calling DeleteResourceExport with out checking backuplocation type.
+	// For other backuplocation type, expecting Notfound
+	crName := getResourceExportCRName(utils.PrefixRestore, string(restore.UID), restore.Namespace)
```
Comment (diptiranjanpx) [inline pkg/applicationmanager/controllers/applicationrestore.go:1700]: Is the intention to delete from both AdminNamespace and restore.Namespace?
Currently both the deletes are deleting from restore.Namespace

Comment (siva-portworx) [reply pkg/applicationmanager/controllers/applicationrestore.go:1700]: Fixed resource's RE also to use restore CR namespace instead of Adminnamespace.

Comment (diptiranjanpx) [inline pkg/applicationmanager/controllers/applicationrestore.go:1700]: We can just loop the same delete steps for both the CRs.

**Review by diptiranjanpx — APPROVED** [review]

File: pkg/applicationmanager/controllers/applicationrestore.go (hunk 1 above)
Comment (siva-portworx) [reply pkg/applicationmanager/controllers/applicationrestore.go]: Fixed

**Pull request merged**